AMP-95395 - add marketing attribution tracking (#242)
amplitude/amplitude-dev-center: docs/data/sdks/browser-2/index.md

amplitude.init(AMPLITUDE_API_KEY, {
  offline: amplitude.Types.OfflineDisabled
});
```

### Marketing Attribution Tracking

> We automatically track marketing attribution by default. Once marketing attribution tracking is enabled, Amplitude will automatically generate identify events to assign the campaign value in certain cases. This ensures that user properties are updated and will influence future events. Below, we outline scenarios that demonstrate when marketing attribution will or will not be tracked. It's important to remember that these examples are illustrative and not exhaustive:

Reviewer: ```suggestion
Amplitude tracks marketing attribution by default. Once you enable marketing attribution tracking, Amplitude generates `identify` events to assign the campaign value in certain cases. This ensures that user properties update and influence future events. 

For more information, see the scenarios outlined below that demonstrate when Amplitude does or doesn't track marketing attribution. These examples are illustrative, not exhaustive:
```

Tracking occurs when either of the following applies:

|Rule|Example|
|-|-|
| A specific referrer domain is explicitly excluded.| When config the `config.defaultTracking.attribution.excludeReferrers` to include `[a.test.com]`, and the page has `a.test.com` as a referrer domain. |
| No previous campaign. | A user's initial visit. |

> | There is an introduction with new UTM parameter or clickId parameter. | If any utm parameters or click id parameters has been droped during a session, we will unset it.|

Reviewer 2: ```suggestion
| There is an introduction of new UTM parameter or `clickId` parameter. | If any utm parameters or click id parameters has been dropped during a session, we will unset it. |
```

| The referrer domain changes to a new one. | Referrer domain changed from `a.test.com` to `b.test-new.com`|

> Conversely, attribution tracking will not be conducted under any of the following conditions:

Reviewer: ```suggestion
Amplitude doesn't track marketing attribution under any of the following conditions:
```

amplitude.init(AMPLITUDE_API_KEY, {
  offline: amplitude.Types.OfflineDisabled
});
```

### Marketing Attribution Tracking

Amplitude tracks marketing attribution by default. Once you enable marketing attribution tracking, Amplitude generates `identify` events to assign the campaign value in certain cases. This ensures that user properties update and influence future events. 
For more information, see the scenarios outlined below that demonstrate when Amplitude does or doesn't track marketing attribution. These examples are illustrative, not exhaustive:

Amplitude tracks marketing attribution by default. Once you enable marketing attribution tracking, Amplitude generates `identify` events to assign the campaign value in certain cases. This ensures that user properties update and influence future events. 

> For more information, see the scenarios outlined below that demonstrate when Amplitude does or doesn't track marketing attribution. These examples are illustrative, not exhaustive.

Reviewer 3: Duplicated?

Tracking occurs when either of the following applies:

|Rule|Example|
|-|-|

> | A specific referrer domain is explicitly excluded.| When config the `config.defaultTracking.attribution.excludeReferrers` to include `[a.test.com]`, and the page has `a.test.com` as a referrer domain. |

Reviewer 3: It looks like this is not a new campaign, `isNewCampaign()` will return false [here ](https://github.com/amplitude/Amplitude-TypeScript/blob/e8216bc65399106214cc5c0ae4d7e8d053029af4/packages/analytics-client-common/src/attribution/campaign-tracker.ts#L55-L57) and will return early [here](https://github.com/amplitude/Amplitude-TypeScript/blob/e8216bc65399106214cc5c0ae4d7e8d053029af4/packages/analytics-client-common/src/attribution/campaign-tracker.ts#L111-L113) before track an identify event.

Author: campaign-tracker.ts has been deprecated. We are not using it in browser sdk.

Reviewer 3: web attribution plugin should have the same logic https://github.com/amplitude/Amplitude-TypeScript/blob/e8216bc65399106214cc5c0ae4d7e8d053029af4/packages/plugin-web-attribution-browser/src/helpers.ts#L34

Reviewer 3: Sync with Alyssa and Alyssa moved this line to examples of not tracking web attribution at https://github.com/amplitude/amplitude-dev-center/pull/242/commits/ffa12b04cf81604cb570973b5182e8122a1ec9b0

| No previous campaign. | A user's initial visit. |

> | There is an introduction of new UTM parameter or `clickId` parameter. | If any utm parameters or `clickId` parameters has been dropped during a session, we will unset it. |

Reviewer 3: Remove code style to keep consistent

https://www.docs.developers.amplitude.com/data/sdks/browser-2/#click-id-parameters
```suggestion
| There is an introduction of new UTM parameter or Click ID parameter. | If any utm parameters or Click ID parameters has been dropped during a session, we will unset it. |
```
